### PR TITLE
Fix URLs

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Mostly HTTP ⤑ HTTPS, but not only.

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [X] GitHub Actions have all passed
* [X] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Comment by DimitriPapadopoulos** [comment]
A linting tool would be nice, but in this case, I started by fixing a single URL and then fixed a few others manually.

**Comment by jakirkham** [comment]
Sphinx has `linkcheck`. This only helps for the files that use Sphinx, but that still may be helpful. That said, sometimes there are timeouts, availability issues, or redirects that can throw these checks

File: zarr/sync.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def __setstate__(self, *args):
 
 class ProcessSynchronizer:
     """Provides synchronization using file locks via the
-    `fasteners <http://fasteners.readthedocs.io/en/latest/api/process_lock.html>`_
+    `fasteners <https://fasteners.readthedocs.io/>`_
```
Comment (joshmoore) [inline zarr/sync.py:29]: https://fasteners.readthedocs.io/en/latest/api/inter_process/ ?

**Comment by joshmoore** [comment]
Thanks. Leaving this open to see if anyone else can help review the URLs.

**Review by joshmoore — APPROVED** [review]
Haven't heard any complaints and I've gone through them all myself. Thanks, @DimitriPapadopoulos!

**Comment by joshmoore** [comment]
Another slow codecov!

**Pull request merged**